plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/AbstractContractStatechartGeneration.xtend

```diff
@@ -77,7 +80,7 @@ abstract class AbstractContractStatechartGeneration {
 
 	def abstract StatechartDefinition execute()
 
-	new(ScenarioDefinition scenario, Component component) {
+	new(ScenarioDeclaration scenario, Component component) {
```

Comment:
Can the `component` not be acquired here inside the constructor using a derived feature? If so, it should not be added here as a separate parameter.

Reply:
At this point of the transformation, we only have the cloned ScenarioDeclaration which is not contained in the ScenarioPackage that holds the reference to the Component. I think that currently it is not possible to find the component of the cloned scenario.